src/array.jl

```diff
@@ -297,7 +297,7 @@ function Base.copyto!(dest::CuArray{T}, doffs::Integer, src::CuArray{T}, soffs::
 end
 
 function Base.unsafe_copyto!(dest::CuArray{T}, doffs, src::Array{T}, soffs, n) where T
-  unsafe_copyto!(pointer(dest, doffs), pointer(src, soffs), n)
+  GC.@preserve src unsafe_copyto!(pointer(dest, doffs), pointer(src, soffs), n)
```

Comment:
The real issue aside, you do need to preserve `dest` here (and a fewe other places) as well assuming `pointer` for `CuArray` and the corresponding `unsafe_copyto!` have a similar semantic as the array one.

Reply:
Yes, trying to copy Array semantics here. So even if returning dest, I need to GC preserve?

Reply:
Yeah, but why waste any brains cells trying to prove you are right, when you are possibly wrong (return values often gets eliminated as a root because of inlining and dead code elimination, turning this into heap smash risk), and the alternative is free and easy?

Reply:
Sure, that's fair. I mainly wanted to validate my mental model, and hadn't considered that.